### ISSUE TITLE
For the downloaded FBX format 3D model package, it cannot be previewed directly in Nextcloud after decompression.

### ISSUE THREAD

**Issue by qiaoqiai0618** [WARP-LAB/files_3dmodelviewer#13]

**Comment by qiaoqiai0618** [comment]
"Failed to import model.(TypeError: Failed to fetch)"

**Comment by kroko** [comment]
pleas use [one of three templates](https://github.com/WARP-LAB/files_3dmodelviewer/issues/new/choose) for issues. i just filled in bug template in your [initial comment](https://github.com/WARP-LAB/files_3dmodelviewer/issues/13#issue-1704988364), please edit it to fill it out.

**Comment by Altair1231** [comment]
Same problem is happening to me,im at Nextcloud 26.0.0
I installed the app via App Store

Just wont load any FBX

**Comment by kroko** [comment]
well, the bugreport quality... I have no idea about any of the required info that OP @MiloGeimer failed to fill, except
- fbx
- via App Store

anyways, I just checked some models. i can replicate issues with FBX, that has embedded textures. for example [this fbx](https://free3d.com/3d-model/sofa-801691.html) works (no textures). does the FBX in the link work for @MiloGeimer and @Altair1231?

it seems that it is upstream issu that I will attend at some point.

**Comment by DevinWinata** [comment]
Hi, I have the same issue and using the [mentioned FBX file](https://free3d.com/3d-model/sofa-801691.html) still can't be previewed.
From browser console logs, it seems CSP is blocking `blob:` when previewing the FBX file.
```
Refused to connect to 'blob:https://example.com/acda3f2d-c3ab-4855-b334-95ebc0af36fa' because it violates the following Content Security Policy directive: "connect-src 'self'".
```
![image](https://github.com/WARP-LAB/files_3dmodelviewer/assets/39239937/225b2296-ccdc-4b11-93f1-0864531bdbc5)
Instance info:
Nextcloud 27 docker (nextcloud with apache built-in docker image `nextcloud:latest` with Caddy as reverse proxy)
Via app store

**Comment by kroko** [comment]
Yes, FBX with embedded textures break badly. I've debugged the issue. Please stand by 😄 ⏳

**Comment by kroko** [comment]
@MiloGeimer, @Altair1231 can you please try 0.0.11.
if it fails, please provide link to exact FBX file that it fails for.

**Comment by ivan4os** [comment]
failed to import model typeerror: failed to fetch (3D Model Viewer app)
files_3dmodelviewer.js?v=13e99c0e-57:1 Refused to connect to 'blob:https://????????/cffe6d08-6aab-48ea-a13c-6bac1da49555' because it violates the following Content Security Policy directive: "connect-src 'self'".
Help me!!!

**Comment by DerLeole** [comment]
I have encountered the same issue and it seems like the csp is actually to blame, since other models load just fine. when they aren't loaded via blob.

**Comment by smarinier** [comment]
@DerLeole @DevinWinata @MiloGeimer @ivan4os
Could you use the "Developer Tools" from your browser (Google Chrome ?), load the NexCloud page, go in "network" tab, and give the whole content of "Content-Security-policy" ?

![DevTools_-_home_smarinier_net_nc28-dev_index_php_apps_files_files_1437_dir___openfile_true](https://github.com/WARP-LAB/files_3dmodelviewer/assets/19839534/14e20239-f738-40ad-9e40-41aa8b971256)

**Comment by DevinWinata** [comment]
@smarinier here's CSP from my nextcloud instance
```
default-src 'none';base-uri 'none';manifest-src 'self';script-src 'nonce-RkU3OVZiR0t5aWpYZ0tJa3E5bm9pRE5PZmM2Y2RVZWRMalJWTTFZSTNkWT06ZlNQSU9Ndk1uMStGNmZaOW5PNkV6SDhoUEpuTU9pLzRWRkFpZVNaaTlvND0=' 'self' drive.example.com 'unsafe-eval';script-src-elem 'strict-dynamic' 'nonce-RkU3OVZiR0t5aWpYZ0tJa3E5bm9pRE5PZmM2Y2RVZWRMalJWTTFZSTNkWT06ZlNQSU9Ndk1uMStGNmZaOW5PNkV6SDhoUEpuTU9pLzRWRkFpZVNaaTlvND0=' 'self' drive.example.com 'unsafe-eval';style-src 'self' 'unsafe-inline';img-src 'self' data: blob: * https://*.tile.openstreetmap.org https://office.example.com;font-src 'self' data:;connect-src 'self';media-src 'self';frame-src 'self' nc: https://office.example.com;frame-ancestors 'self' https://office.example.com;worker-src 'self';form-action 'self' https://office.example.com
```

**Comment by smarinier** [comment]
@DevinWinata could you try to (temporarily) install and active the "Talk" application in NextCloud. And tell me if it's ok now for the FBX preview ?

**Comment by DevinWinata** [comment]
@smarinier hi, it's now able to preview FBX files.

**Comment by smarinier** [comment]
Ok. I see what the fix could be.

**Comment by smarinier** [comment]
Hi @kroko,

Could you have a look on the very simple PR #26 ?

Thanks

**Comment by kroko** [comment]
@smarinier merged and pushed as a new release v0.0.14 in NC appstore, thanks! wait till it lands and test it out.
while first getting this plugin up and running CSP was headache, especially because for some file type loaders it [has to lazyload extra js and wasm](https://github.com/WARP-LAB/files_3dmodelviewer/blob/ed457db5c1dcd43aee637e5d97e16d7263e7bf01/src/js/App/App.mjs#L351) however based on what plugins user has installed (i specially installed A LOT in the dev instance) the whole CSP policy goes haywire.
